h/views/api/annotations.py

```diff
@@ -190,18 +190,31 @@ def _annotation_resource(request, annotation):
 
 def _record_search_api_usage_metrics(
     params,
-    record_metrics=newrelic.agent.record_custom_metrics,
+    record_param=newrelic.agent.add_custom_parameter,
 ):
-    metrics = [
-        # Record usage of deprecated offset.
-        ('Custom/SearchApi/offset', int("offset" in params)),
-
+    # Record usage of search params and associate them with a transaction.
+    keys = [
+        # Record usage of inefficient offset and it's alternative search_after.
+        "offset",
+        "search_after",
+        "sort",
         # Record usage of url/uri (url is an alias of uri).
-        ('Custom/SearchApi/url', int("url" in params)),
-        ('Custom/SearchApi/uri', int("uri" in params)),
-
+        "url",
+        "uri",
         # Record usage of tags/tag (tags is an alias of tag).
-        ('Custom/SearchApi/tags', int("tags" in params)),
-        ('Custom/SearchApi/tag', int("tag" in params)),
+        "tags",
+        "tag",
+        # Record usage of _separate_replies which will help distinguish client calls
+        # for loading the sidebar annotations from other api calls.
+        "_separate_replies",
+        # Record group and user-these help in identifying slow queries.
+        "group",
+        "user",
+        # Record usage of wildcard feature.
+        "wildcard_uri",
     ]
-    record_metrics(metrics)
+
+    for k in keys:
+        if k in params:
+            # The NewRelic query language does not permit _ at the begining.
+            record_param(k.lstrip('_'), str(params[k]))
```

Comment:
The "tags" param is a list here isn't it? If so that means that searches for `["foo"]`, `["foo", "bar"]`, and `["bar", "foo"]` will presumably count separately (rather than as three searches involving tag "foo"? That might be what you wanted, but thought I'd raise the question.

Reply:
Yes, I think that's fine.